php-checks/src/main/java/org/sonar/php/checks/HardCodedCredentialsInFunctionCallsCheck.java

```diff
@@ -0,0 +1,223 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.ArgumentMatcher;
+import org.sonar.php.checks.utils.ArgumentVerifierUnaryFunction;
+import org.sonar.php.checks.utils.CheckUtils;
+import org.sonar.php.checks.utils.FunctionArgumentCheck;
+import org.sonar.php.tree.impl.declaration.ClassNamespaceNameTreeImpl;
+import org.sonar.plugins.php.api.tree.Tree.Kind;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.MemberAccessTree;
+import org.sonar.plugins.php.api.tree.expression.NewExpressionTree;
+import org.sonarsource.analyzer.commons.internal.json.simple.JSONArray;
+import org.sonarsource.analyzer.commons.internal.json.simple.JSONObject;
+import org.sonarsource.analyzer.commons.internal.json.simple.parser.JSONParser;
+import org.sonarsource.analyzer.commons.internal.json.simple.parser.ParseException;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+@Rule(key = "S6437")
+public class HardCodedCredentialsInFunctionCallsCheck extends FunctionArgumentCheck {
+
+  private static final String MESSAGE = "Revoke and change this password, as it is compromised.";
+
+  private static final String JSON_LOCATION = "HardCodedCredentialsSensitiveFunctions.json";
+
+  private static final Map<String, SensitiveMethod> SENSITIVE_FUNCTIONS = JsonSensitiveFunctionsReader.parseSensitiveFunctions();
+
+  private static final Logger LOG = Loggers.get(HardCodedCredentialsInFunctionCallsCheck.class);
+
+  @Override
+  public void visitNewExpression(NewExpressionTree tree) {
+    FunctionCallTree functionCallTree = (FunctionCallTree) tree.expression();
+    ClassNamespaceNameTreeImpl callee = (ClassNamespaceNameTreeImpl) functionCallTree.callee();
+    String fqn = callee.symbol().qualifiedName().toString();
```

Comment:
```suggestion
    QualifiedName fqn = callee.symbol().qualifiedName();
```
You don't need to call `toString` when only used in a concatenation. This would make it more readable.